x-pack/plugins/apm/server/routes/services/get_services_detailed_statistics/get_service_aggregated_transaction_detailed_statistics.ts

```diff
@@ -141,13 +141,17 @@ export async function getServiceAggregatedTransactionDetailedStats({
   );
 
   return keyBy(
-    response.aggregations?.sample.services.buckets.map((bucket) => {
+    response.aggregations?.sample.services.buckets.forEach((bucket) => {
       const topTransactionTypeBucket =
         bucket.transactionType.buckets.find(
           ({ key }) =>
             key === TRANSACTION_REQUEST || key === TRANSACTION_PAGE_LOAD
         ) ?? bucket.transactionType.buckets[0];
 
+      if(!topTransactionTypeBucket) {
+        return [];
+      }
```

Comment:
Sure, as I mentioned in the ticket I was able to reproduce this bug only with synthrace but when I generate the data it indexes only system/app metrics but no trace data. 

Since there are no trace data , `topTransactionTypeBucket` is undefined thus accessing any property of it will fail with an error. 

....or it can be solved with using optional chaining operator

Reply:
Why does it return buckets if there are no matching documents for the given query?